### PR TITLE
Renamed src/physics/common -> src/physics/share, moved saturation tests to src/physics/share/tests

### LINKED ISSUE #466: Move saturation unit tests from physics/p3/tests to physics/share/tests
Right now, saturation calculations are performed in `physics/common/`, but unit tests for saturation functions are in `physics/p3/tests/`. This is inconsistent and should be fixed. While we're at it, we can rename `physics/common` to `physics/share/` to emulate naming already used in SCREAM. 

This task involves:

1. Renaming the `physics/common/` directory to `physics/shared` and updating references within the build system and code.
2. Creating a new `physics/share/tests directory that contains share_tests.cpp and share_unit_tests_common.cpp files.
3. Migrating the saturation unit tests to the `physics/share/tests` directory.

### PR DESCRIPTION
**Pull request by jeff-cohere**
Closes #466

### CODE REVIEW

File: components/cam/src/physics/cam/bfb_math.inc [hunk 1]
```diff
@@ -6,8 +6,8 @@
 ! Make sure to place the following lines at the top of any modules
 ! that use these macros:
 !
-! use physics_common_iso_f, only: cxx_pow, cxx_sqrt, cxx_cbrt, cxx_gamma, cxx_log, &
-!                                 cxx_log10, cxx_exp
+! use physics_share_iso_f, only: cxx_pow, cxx_sqrt, cxx_cbrt, cxx_gamma, cxx_log, &
```
Comment (bartgol) [inline components/cam/src/physics/cam/bfb_math.inc:9]: Is it too invasive of a modification to add the change from `_iso_f` to `_f2c`, as we discussed somewhere (maybe confluence)? Imho, it's perfectly ok to leave that change for another PR, but since you were renaming things...

Comment (jeff-cohere) [reply components/cam/src/physics/cam/bfb_math.inc:9]: I can do that.

File: components/scream/src/physics/share/tests/physics_unit_tests_common.hpp [hunk 2]
```diff
@@ -0,0 +1,65 @@
+#ifndef PHYSICS_UNIT_TESTS_COMMON_HPP
```
Comment (PeterCaldwell) [inline components/scream/src/physics/share/tests/physics_unit_tests_common.hpp:1]: should this file be named share instead of common? 8-)

Comment (jeff-cohere) [reply components/scream/src/physics/share/tests/physics_unit_tests_common.hpp:1]: Yes.

Comment (jeff-cohere) [reply components/scream/src/physics/share/tests/physics_unit_tests_common.hpp:1]: Actually, no. The COMMON part comes from the fact that these constructs are provided as a common element to all physics/share unit tests. This mirrors src/physics/p3/tests/p3_unit_tests_common.hpp, for example.

Comment (jeff-cohere) [reply components/scream/src/physics/share/tests/physics_unit_tests_common.hpp:1]: Naming things is hard.

**Comment by jeff-cohere** [comment]
> Looks great. Have you recursive-grepped for common in our scream directories to make sure you caught them all?

Yeah, I think so. I didn't do a clean build last time, which is why I missed the Fortran stuff. Bad developer!

File: components/scream/src/physics/share/tests/physics_unit_tests_common.hpp (hunk 2 above)
Comment (PeterCaldwell) [reply components/scream/src/physics/share/tests/physics_unit_tests_common.hpp:1]: Yeah, I hear you. Ok, I'm fine w/ keeping common.
